cuda_core/docs/design/buffer-fill-redesign.md

```diff
@@ -0,0 +1,77 @@
+# Buffer.fill() Redesign Proposal
+
+**Issue**: [#1345 - Revisit `Buffer.fill()`](https://github.com/NVIDIA/cuda-python/issues/1345)  
+**Author**: Andy Jost  
+**Date**: December 10, 2025
+
+## Background
+
+PR #1318 implemented `Buffer.fill(value, width, *, stream)` but was merged before review feedback was addressed. This document proposes a simplified API based on that feedback.
+
+## Proposed API
+
+```python
+def fill(self, value, *, stream: Stream | GraphBuilder):
+    """Fill buffer with a repeating byte pattern.
+    
+    Parameters
+    ----------
+    value : int or buffer-protocol object
+        - int: Must be in range [0, 256). Converted to 1 byte.
+        - buffer-protocol object: Must be 1, 2, or 4 bytes.
```

Comment:
super nitpick, but there's the `collections.abc.Buffer` type that exists specifically to indicate objects that support buffer protocol